src/controllers/DrawingModeController.js

```diff
@@ -431,6 +429,19 @@ class DrawingModeController extends AnnotationModeController {
             }
         }
     }
+
+    /**
+     * Save the original thread and create a new thread
+     *
+     * @private
+     * @param {AnnotationThread} thread The thread that emitted the event
+     * @return {void}
+     */
+    saveCurrentThread(thread) {
+        this.currentThread = undefined;
+        thread.saveAnnotation(TYPES.draw);
+        this.registerThread(thread);
```

Comment:
Is it possible to make sure that we've successfully saved before nullifying `this.currentThread`? Or does it not matter if anything goes wrong with the state of the thread, if it fails to save?

Reply:
Fixed this. We should only be marking the currentThread as undefined when thread is explicitly saved